### PR TITLE
update interface to share all streams between all sockets

### PR DESCRIPTION
**Pull request by mafintosh**

### CODE REVIEW

File: lib/socket.js [hunk 1]
```diff
@@ -4,11 +4,12 @@ const events = require('events')
...
 const IPv4Pattern = /^((?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])[.]){3}(?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$/
 
-module.exports = class UDX extends events.EventEmitter {
-  constructor () {
+module.exports = class UDXSocket extends events.EventEmitter {
+  constructor (udxHandle) {
     super()
 
-    this._handle = Buffer.allocUnsafe(binding.sizeof_udx_napi_t)
+    this._udxHandle = udxHandle
```
Comment (kasperisager) [inline lib/socket.js:11]: Is the `Handle` postfix helpful?

Comment (kasperisager) [reply lib/socket.js:11]: Ah, it passes the actual handle, not the JavaScript class instance. Would `_udx._handle` be better?

Comment (mafintosh) [reply lib/socket.js:11]: wanna push that fix?

**Comment by mafintosh** [comment]
@kasperisager good to merge or do you have anything pending we should change?

**Comment by kasperisager** [comment]
@mafintosh Nope, let's ship it 🚀

**Comment by mafintosh** [comment]
💪

**Pull request merged**